client/reader/search-stream/suggestion-provider.jsx

```diff
@@ -19,36 +20,75 @@ import { getReaderFollowedTags } from 'state/selectors';
  * @return {Array}       An array of suggestions, or null if no tags where provided
  */
 function suggestionsFromTags( count, tags ) {
+	const tagSuggestions = [];
+	let sampleSuggestions = {};
 	if ( tags ) {
 		if ( tags.length <= count ) {
 			return [];
 		}
-		return map( sampleSize( tags, count ), tag => ( tag.displayName || tag.slug ).replace( /-/g, ' ' ) );
+		sampleSuggestions = map( sampleSize( tags, count ), tag => ( tag.displayName || tag.slug ).replace( /-/g, ' ' ) );
+
+		for ( let i = 0; i < sampleSuggestions.length; i++ ) {
```

Comment:
what is ui_position for?

Because you reuse the same algo for generating this in two places, it would also be better to create a  `mapTagToSuggestion( tag, i ) --> suggestionWithRailcar` function